packages/flutter_tools/lib/src/commands/migrate_apply.dart

```diff
@@ -0,0 +1,197 @@
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    logger.printStatus('Applying migration.');
+    // Copy files from working directory to project root
+    final List<String> allFilesToCopy = <String>[];
+    allFilesToCopy.addAll(manifest.mergedFiles);
+    allFilesToCopy.addAll(manifest.conflictFiles);
+    allFilesToCopy.addAll(manifest.addedFiles);
+    if (allFilesToCopy.isNotEmpty && _verbose) {
+      logger.printStatus('Modifying ${allFilesToCopy.length} files.', indent: 2);
+    }
+    for (final String localPath in allFilesToCopy) {
+      if (_verbose) {
+        logger.printStatus('Writing $localPath');
+      }
+      final File workingFile = workingDirectory.childFile(localPath);
+      final File targetFile = project.directory.childFile(localPath);
+      if (!workingFile.existsSync()) {
+        continue;
+      }
+
+      if (!targetFile.existsSync()) {
+        targetFile.createSync(recursive: true);
+      }
+      try {
+        targetFile.writeAsStringSync(workingFile.readAsStringSync(), flush: true);
+      } on FileSystemException {
+        targetFile.writeAsBytesSync(workingFile.readAsBytesSync(), flush: true);
+      }
+    }
+    // Delete files slated for deletion.
+    if (manifest.deletedFiles.isNotEmpty) {
+      logger.printStatus('Deleting ${manifest.deletedFiles.length} files.', indent: 2);
+    }
+    for (final String localPath in manifest.deletedFiles) {
+      final File targetFile = FlutterProject.current().directory.childFile(localPath);
+      targetFile.deleteSync();
+    }
+
+    // Update the migrate config files to reflect latest migration.
+    if (_verbose) {
+      logger.printStatus('Updating .migrate_configs');
+    }
+    final FlutterProjectMetadata metadata = FlutterProjectMetadata(project.directory.childFile('.metadata'), logger);
+    final FlutterVersion version = FlutterVersion(workingDirectory: project.directory.absolute.path);
+
+    final String currentGitHash = version.frameworkRevision;
+    metadata.migrateConfig.populate(
+      projectDirectory: project.directory,
+      currentRevision: currentGitHash,
+      logger: logger,
+    );
+
+    // Clean up the working directory
+    final bool keepWorkingDirectory = boolArg('keep-working-directory') ?? false;
+    if (!keepWorkingDirectory) {
+      workingDirectory.deleteSync(recursive: true);
+    }
+
+    // Detect pub dependency locking. Run flutter pub upgrade --major-versions
+    await updatePubspecDependencies(project, migrateUtils, logger, terminal);
+
+    // Detect gradle lockfiles in android directory. Delete lockfiles and regenerate with ./gradlew tasks (any gradle task that requires a build).
+    await updateGradleDependencyLocking(project, migrateUtils, logger, terminal, _verbose, fileSystem);
+
+    logger.printStatus('Migration complete. You may use commands like `git status`, `git diff` and `git restore <file>` to continue working with the migrated files.');
```

Comment:
nit: line is long.  https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters